kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -204,7 +214,8 @@ public Transaction build(Engine engine) {
         shouldUpdateProtocol,
         maxRetries,
         table.getClock(),
-        getDomainMetadatasToCommit(snapshot));
+        getDomainMetadatasToCommit(snapshot),
+        !isNewTable && updatedSchema);
```

Comment:
Can you comment what this parameter is preserveFieldIds

Reply:
Sure, I'll add an inline comment!